mypy/main.py

```diff
@@ -555,18 +558,12 @@ def create_source_list(files: Sequence[str], options: Options) -> List[BuildSour
     targets = []
     for f in files:
         if f.endswith(PY_EXTENSIONS):
-            try:
-                targets.append(BuildSource(f, crawl_up(f)[1], None))
-            except InvalidPackageName as e:
-                fail(str(e))
+            targets.append(BuildSource(f, crawl_up(f)[1], None))
         elif os.path.isdir(f):
-            try:
-                sub_targets = expand_dir(f)
-            except InvalidPackageName as e:
-                fail(str(e))
+            sub_targets = expand_dir(f)
             if not sub_targets:
-                fail("There are no .py[i] files in directory '{}'"
-                     .format(f))
+                raise Exception("There are no .py[i] files in directory '{}'"
```

Comment:
Please define a custom exception for this purpose. `Exception` is for catching, not throwing. (I see that this also throws `InvalidPackageName`, from `crawl_up()`, maybe you can rename and reuse it?